teps/0050-ignore-task-failures.md

```diff
@@ -0,0 +1,102 @@
+    "message": "Tasks Completed: 1 (Failed: 1, Cancelled 0), Skipped: 3",
+    "reason": "Failed",
+    "status": "False",
+    "type": "Succeeded"
+  }
+]
+```
+
+Tekton [catalog](https://github.com/tektoncd/catalog) has a wide range of `tasks` which are designed to be reusable
+in many pipelines. As a pipeline execution engine, we encourage the pipeline authors to utilize arbitrary tasks from
+the Tekton catalog. But, many common pipelines have the requirement where a task failure must not block executing the
+rest of the tasks.
+
+A pipeline author has an option to utilize `finally` section of the pipeline in which all the final tasks are executed
+after all the tasks in a graph have completed regardless of success or failure. `finally` has its own advantages and
+very helpful in various use cases including notifications, cleanup, etc.
+
+But, the pipeline authors does not have the flexibility to unblock executing the rest of the graph after experiencing a
+single task failure.
+
+
+## Motivation
+
+It should be possible to utilize tasks from the Tekton catalog in a pipeline. A pipeline author has no
+control over the task definitions but may desire to ignore a failure and continue executing the rest of the graph.
+
+
+### Goals
+
+* Design a task failure strategy so that the pipeline author can control the behavior of the underlying tasks 
+  and decide whether to continue executing the rest of the graph in the event of failure.
+
+* Be applicable to any pipeline with references to the tasks in a catalog or inlined task specifications.
+
+### Non-Goals
+
+* Not an alternative to combining the tasks in a pipeline which is covered in
+  [TEP-0044 Composing Tasks with Tasks](https://github.com/tektoncd/community/pull/316).
+
+* Not optimizing pipeline runtime which is covered in
+  [TEP-0046 PipelineRun in a Pod](https://github.com/tektoncd/community/pull/318).
+
+## Requirements
+
+* Users should be able to use any task from the catalog without having to alter its specification to allow that task to
+  fail without stopping the execution of a pipeline.
+
+* It should be possible to know that a task failed, and the rest of the graph was allowed to continue by observing
+  the status of the `PipelineRun`.
+
+
+### Use Cases
+
+* As a pipeline author, I would like to design a pipeline where a task running
+  [unit tests](https://github.com/tektoncd/catalog/tree/master/task/golang-test/0.1) might fail,
+  but can continue running integration tests, so that my pipeline can identify failures in both the tests.
+
+* As a pipeline author, I would like to design a pipeline where a task running
+  [linting](https://github.com/tektoncd/catalog/tree/master/task/golangci-lint/0.1) might fail,
+  but can continue running tests, so that my pipeline can report failures from the linting and all the tests.
+
+* As a new Tekton user, I want to migrate existing workflows from the other CI/CD systems that allowed a
+  similar task unit of failure.
```

Comment:
is it possible to give some examples? i think a lot of systems that folks are migrating from dont have the same parallelism that Pipelines support, so I'm wondering if that might be the answer to most of these cases

Reply:
Maybe we can include an example of a Jenkinsfile (with a failure allowed) that we'd be migrating? I'm struggling with trying to understand if one of the main motivations around the migration use case is that many CD systems do not allow for parallel execution, so it'd be great to see a counter example

Reply:
sure, I can add an example, there are other tools as well, Travis, Argo Workflow, etc